src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -260,89 +261,139 @@ Status ServiceBasedActorInfoAccessor::AsyncSubscribe(
                  << ", job id = " << actor_id.JobId();
   RAY_CHECK(subscribe != nullptr) << "Failed to subscribe actor, actor id = " << actor_id;
 
-  auto fetch_data_operation = [this, actor_id,
-                               subscribe](const StatusCallback &fetch_done) {
-    auto callback = [actor_id, subscribe, fetch_done](
-                        const Status &status,
-                        const boost::optional<rpc::ActorTableData> &result) {
-      if (result) {
-        subscribe(actor_id, *result);
-      }
-      if (fetch_done) {
-        fetch_done(status);
-      }
+  if (grpc_based_pubsub_) {
+    auto &subscriber = client_impl_->GetGcsSubscriber();
+    auto sub_message = std::make_unique<rpc::SubMessage>();
+    rpc::Address gcs_address;
+    gcs_address.set_ip_address(client_impl_->GetGcsServerAddress().first);
+    gcs_address.set_port(client_impl_->GetGcsServerAddress().second);
+    auto subscription_callback = [actor_id, subscribe](const rpc::PubMessage &message) {
+      RAY_LOG(ERROR) << "Got a subscription!";
+      RAY_CHECK(message.channel_type() == rpc::ChannelType::GCS_ACTOR_CHANNEL);
+      RAY_CHECK(message.has_actor_table_data());
+      ActorID id = ActorID::FromBinary(message.actor_table_data().actor_id());
+      RAY_CHECK(id == actor_id);
+      subscribe(id, message.actor_table_data());
+    };
+    auto failure_callback = [done](const std::string &message) {
+      done(Status::UnknownError(message));
+    };
+    subscriber.Subscribe(std::move(sub_message), rpc::ChannelType::GCS_ACTOR_CHANNEL,
+                         gcs_address, actor_id.Binary(), subscription_callback,
+                         failure_callback);
+    return Status::OK();
+  } else {
```

Comment:
Can you un-indent the else block? That would avoid such a huge diff below which presumably hasn't changed. There's an early return above so it's not needed.